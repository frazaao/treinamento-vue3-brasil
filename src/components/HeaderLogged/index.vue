<template>
  <div class="flex justify-center w-full h-28 bg-brand-main">
    <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
      <picture class="w-28 lg:w-36">
        <img
          class="w-full"
          src="../../assets/images/logo_white.png"
          alt="Logomarca"
        />
      </picture>

      <nav>
        <ul class="flex list-none">
          <li
            @click="() => router.push({ name: 'Credentials' })"
            class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
          >
            Credenciais
          </li>

          <li
            @click="() => router.push({ name: 'Feedbacks' })"
            class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
          >
            Feedbacks
          </li>

          <li
            class="px-6 py-2 mr-2 font-bold bg-white text-brand-main rounded-full cursor-pointer focus:outline-none"
            @click="handleLogout"
          >
            {{ logoutLabel }}
          </li>
        </ul>
      </nav>
    </div>
  </div>
</template>

<script>
import { useRouter } from "vue-router";
import { computed } from "vue";
import useStore from "../../hooks/useStore";
import { clearCurrentUser } from "../../store/user";

export default {
  name: "HeaderLogged",
  setup() {
    const router = useRouter();
    const store = useStore("User");

    const logoutLabel = computed(() => {
      if (!store.currentUser.name) {
        return "...";
      } else {
        return store.currentUser.name + " (sair)";
      }
    });

    function handleLogout() {
      window.localStorage.removeItem("token");
      clearCurrentUser();
      router.push({ name: "Home" });
    }

    return {
      router,
      logoutLabel,
      handleLogout,
    };
  },
};
</script>

<style lang="scss" scoped></style>
