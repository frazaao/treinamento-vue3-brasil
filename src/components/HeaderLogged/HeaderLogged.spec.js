import HeaderLogged from "./index.vue";
import { mount } from "@vue/test-utils";
import { routes } from "@/router";
import { createRouter, createWebHistory } from "vue-router";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

const mockStore = {
  currentUser: {},
};

jest.mock("../../hooks/useStore", () => {
  return () => {
    return mockStore;
  };
});

describe("<HeaderLogged />", () => {
  it("Should render header logged correctly", async () => {
    router.push("/");

    await router.isReady();

    const wrapper = mount(HeaderLogged, {
      global: { plugins: [router] },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should render 3 dots when there is no user logged in", async () => {
    router.push("/");

    await router.isReady();

    const wrapper = mount(HeaderLogged, {
      global: { plugins: [router] },
    });

    const buttonLogout = wrapper.find("#logout-button");

    expect(buttonLogout.text()).toBe("...");
  });

  it("Should render username when there is no user logged in", async () => {
    router.push("/");

    mockStore.currentUser.name = "Matheus";

    await router.isReady();

    const wrapper = mount(HeaderLogged, {
      global: { plugins: [router] },
    });

    const buttonLogout = wrapper.find("#logout-button");

    expect(buttonLogout.text()).toBe("Matheus (sair)");
  });
});
