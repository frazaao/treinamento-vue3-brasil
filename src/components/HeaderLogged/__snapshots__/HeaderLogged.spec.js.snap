// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> Should render header logged correctly 1`] = `
<div class="flex justify-center w-full h-28 bg-brand-main">
  <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
    <picture class="w-28 lg:w-36"><img class="w-full" src="" alt="Logomarca"></picture>
    <nav>
      <ul class="flex list-none">
        <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credenciais </li>
        <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
        <li class="px-6 py-2 mr-2 font-bold bg-white text-brand-main rounded-full cursor-pointer focus:outline-none" id="logout-button">...</li>
      </ul>
    </nav>
  </div>
</div>
`;
